internal/cgroups/cgroups.go

```diff
@@ -115,3 +142,69 @@ func (cg CGroups) CPUQuota() (float64, bool, error) {
 
 	return float64(cfsQuotaUs) / float64(cfsPeriodUs), true, nil
 }
+
+// IsCGroupV2 returns true if the system supports and uses cgroup2.
+// It gets the required information for deciding from mountinfo file.
+func IsCGroupV2() (bool, error) {
+	return isCGroupV2(_procPathMountInfo)
+}
+
+func isCGroupV2(procPathMountInfo string) (bool, error) {
+	var isV2 bool
+	newMountPoint := func(mp *MountPoint) error {
+		if mp.FSType == _cgroupv2FSType && mp.MountPoint == _cgroupv2MountPoint {
+			isV2 = true
+		}
+		return nil
+	}
+	if err := parseMountInfo(procPathMountInfo, newMountPoint); err != nil {
+		return false, err
+	}
+	return isV2, nil
+}
+
+// CPUQuotaV2 returns the CPU quota applied with the CPU cgroup2 controller.
+// It is a result of reading cpu quota and period from cpu.max file.
+// It will return `cpu.max / cpu.period`. If cpu.max is set to max, it returns
+// (-1, false, nil)
+func CPUQuotaV2() (float64, bool, error) {
+	return cpuQuotaV2(_cgroupv2MountPoint, _cgroupv2CPUMax)
+}
+
+func cpuQuotaV2(cgroupv2MountPoint, cgroupv2CPUMax string) (float64, bool, error) {
+	cpuMaxParams, err := os.Open(path.Join(cgroupv2MountPoint, cgroupv2CPUMax))
```

Comment:
Here you should also include the cgroup-path from `/proc/self/cgroup`, see https://man7.org/linux/man-pages/man7/cgroups.7.html